community/cypher/cypher-logical-plans-3.4/src/main/scala/org/neo4j/cypher/internal/v3_4/logical/plans/LogicalPlan.scala

```diff
@@ -37,55 +38,32 @@ non-leaf nodes are algebraic operators like selections, projections, and joins.
 application of the corresponding operator on the relations generated by its children, the result of which is then sent
 further up. Thus, the edges of a tree represent data flow from bottom to top, i.e., from the leaves, which correspond
 to data in the database, to the root, which is the final operator producing the query answer. */
-abstract class LogicalPlan
+abstract class LogicalPlan(idGen: IdGen)
   extends Product
   with Foldable
   with Strictness
-  with RewritableWithMemory {
+  with Rewritable {
 
   self =>
 
+  // FIXME this should be removed ASAP
```

Comment:
Maybe add to the comment that fact that the bug has been reported and fixed, and when we can expect to be able to remove it?

Reply:
Comment improved